ed25519/src/impl_signature_trait.rs

```diff
@@ -0,0 +1,82 @@
+pub mod signers {
+    //! [`libcrux_traits::signature`] APIs.
+
+    use libcrux_secrets::{DeclassifyRef, U8};
+    use libcrux_traits::signature::arrayref::{KeyGenError, Sign, SignError, VerifyError};
+
+    /// A convenience struct for signature scheme functionality.
+    pub struct Signer;
+
+    const VERIFICATION_KEY_LEN: usize = 32;
+    const SIGNING_KEY_LEN: usize = 32;
+    const SIGNATURE_LEN: usize = 64;
+    const RAND_KEYGEN_LEN: usize = SIGNING_KEY_LEN;
+
+    impl Sign<SIGNING_KEY_LEN, VERIFICATION_KEY_LEN, SIGNATURE_LEN, RAND_KEYGEN_LEN> for Signer {
+        /// Sign a payload with a provided signing key.
+        fn sign(
+            payload: &[u8],
+            signing_key: &[U8; SIGNING_KEY_LEN],
+            signature: &mut [u8; SIGNATURE_LEN],
+            _aux: (),
```

Comment:
Having a useless parameter in the API isn't great. Maybe the `aux` version should be a different trait? Or maybe traits just make it more complicated than necessary for signatures.